mail/spring-cloud-starter-stream-source-mail/src/main/java/org/springframework/cloud/stream/app/mail/source/MailSourceProperties.java

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.stream.app.mail.source;
+
+import java.util.Properties;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+/**
+ * Configuration properties for the Mail Source module.
+ *
+ * @author Amol
+ */
+@ConfigurationProperties
+public class MailSourceProperties {
```

Comment:
@sobychacko , thank you for your attention! 
I've seen some docs today already, but right now... @aamol , can you explain where is that `README` ?

Regarding `mail-url` and `protocol`. Looks weird. 
I really sure that we can extract protocol from the URL. Similar like `AbstractMailReceiver` does that:
```
public AbstractMailReceiver(String url) {
		if (url != null) {
			this.url = new URLName(url);
		}
		else {
			this.url = null;
		}
	}
```
and afterwards:
```
this.url.getProtocol();
```